node/src/components/consensus/consensus_protocol.rs

```diff
@@ -200,4 +200,7 @@ pub(crate) trait ConsensusProtocol<I, C: Context>: Send {
 
     /// Returns the instance ID of this instance.
     fn instance_id(&self) -> &C::InstanceId;
+
+    /// Returns the protocol outcomes for all the required timers.
+    fn recreate_timers(&self) -> Vec<ProtocolOutcome<I, C>>;
```

Comment:
If possible, I would prefer to not expose this method in the API of the `ConsensusProtocol`. It's leaking implementation details (namely timers) to the clients.

Reply:
We could call it something more generic (`reinitialize_component` or something?), but otherwise I think this is where it belongs - if we ever used other consensus protocols, they would also have to implement such a method in order to be working correctly after the transition to validator.

Reply:
That's my point – it's not required that all consensus protocols would use timers but I am fine leaving it here for now (with a rename) and remove the `EraSupervisor` form the `joiner` later.